tensornetwork/block_sparse/charge.py

```diff
@@ -114,3 +114,272 @@ def __repr__(self):
 
   def __len__(self):
     return len(self.charge_labels)
+
+  def __eq__(self,
+             target_charges: Union[np.ndarray, "BaseCharge"]) -> np.ndarray:
+
+    if isinstance(target_charges, type(self)):
+      targets = np.unique(
+          target_charges.unique_charges[:, target_charges.charge_labels],
+          axis=1)
+    else:
+      if target_charges.ndim == 1:
+        target_charges = np.expand_dims(target_charges, 0)
+      targets = np.unique(target_charges, axis=1)
+    #pylint: disable=no-member
+    inds = np.nonzero(
+        np.logical_and.reduce(
+            np.expand_dims(self.unique_charges,
+                           2) == np.expand_dims(targets, 1),
+            axis=0))[0]
+    return np.expand_dims(self.charge_labels, 1) == np.expand_dims(inds, 0)
+
+  @property
+  def identity_charges(self) -> "BaseCharge":
+    """
+    Returns the identity charge.
+    Returns:
+      BaseCharge: The identity charge.
+    """
+    unique_charges = np.expand_dims(
+        np.asarray([ct.identity_charge() for ct in self.charge_types],
+                   dtype=np.int16), 1)
+    charge_labels = np.zeros(1, dtype=np.int16)
+    obj = self.__new__(type(self))
+    obj.__init__(unique_charges, charge_labels, self.charge_types)
+    return obj
+
+  def __add__(self, other: "BaseCharge") -> "BaseCharge":
+    """
+    Fuse `self` with `other`.
+    Args:
+      other: A `BaseCharge` object.
+    Returns:
+      BaseCharge: The result of fusing `self` with `other`.
+    """
+
+    # fuse the unique charges from each index, then compute new unique charges
+    comb_charges = fuse_ndarray_charges(self.unique_charges,
+                                        other.unique_charges, self.charge_types)
+    [unique_charges, charge_labels] = np.unique(
+        comb_charges, return_inverse=True, axis=1)
+    charge_labels = charge_labels.reshape(self.unique_charges.shape[1],
+                                          other.unique_charges.shape[1]).astype(
+                                              np.int16)
+
+    # find new labels using broadcasting
+    charge_labels = charge_labels[(
+        self.charge_labels[:, None] +
```

Comment:
Can you break this into multiple statements? It's hard to read currently

Reply:
fixed